internal/app/machined/internal/phase/rootfs/check_install.go

```diff
@@ -32,6 +32,10 @@ func (task *CheckInstall) TaskFunc(mode runtime.Mode) phase.TaskFunc {
 }
 
 func (task *CheckInstall) standard(r runtime.Runtime) (err error) {
-	_, err = os.Stat(filepath.Join(constants.BootMountPoint, "installed"))
+	_, err = metadata.Open(filepath.Join(constants.BootMountPoint, constants.MetadataFile))
```

Comment:
The `metadata.Open` function has an additional check to ensure that the file is not empty.